src/ralph/ssl_certificates/management/commands/update_service_env.py

```diff
@@ -0,0 +1,82 @@
+import requests
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from ralph.assets.models.assets import ServiceEnvironment
+from ralph.dns.dnsaas import dnsaas_client
+from ralph.ssl_certificates.models import SSLCertificate
+
+
+def checking_type(value):
+    if value['type'] in {'CNAME', 'PTR'}:
+        domain = value['content']
+    else:
+        domain = value['name']
+    return domain
+
+
+class Command(BaseCommand):
+    help = 'Checks the compliance of services in SSL Certificates'
+
+    def get_records(self):
+        url = dnsaas_client.build_url('records')
+        response = requests.get(
+            '{}'.format(url), headers={
+                'Authorization': 'Token {}'.format(settings.DNSAAS_TOKEN)
+            })
+        limit = response.json()['count']
+        record_limit = range(0, limit, 50)
+        for next in record_limit:
+            url = dnsaas_client.build_url('records', get_params=[
+                ('limit', 50),
+                ('offset', next)
+            ])
+            response = requests.get(
+                '{}'.format(url), headers={
+                    'Authorization': 'Token {}'.format(settings.DNSAAS_TOKEN)
+                })
+            results = response.json()['results']
+            self.update_from_record(results)
+
+    def update_from_record(self, result):
+        for value in result:
+            domain = checking_type(value)
+            service_dns = value['service_name']
+            try:
+                service_env = ServiceEnvironment.objects.get(
+                    service__name=service_dns, environment__name='prod'
+                )
+            except ServiceEnvironment.DoesNotExist:
+                self.stderr.write(
+                    'Service with name {} does not exist'.format(service_dns)
+                )
+            else:
+                SSLCertificate.objects.filter(
+                    domain_ssl=domain).update(
+                    service_env=service_env)
+
+    def update_from_domains(self, result):
+        for value in result:
+            domain = value['name']
+            service_dns = value['service_name']
+            try:
+                name = ServiceEnvironment.objects.get(
+                    service__name=service_dns, environment__name='prod'
+                )
+            except ServiceEnvironment.DoesNotExist:
+                self.stderr.write(
+                    'Service with name {} does not exist'.format(service_dns)
+                )
+            else:
+                SSLCertificate.objects.filter(
```

Comment:
maybe move this operation to separate method? And then change also line 54

Reply:
done